src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/Backup.java

```diff
@@ -0,0 +1,579 @@
+        Backup.context = context;
+
+        try {
+            AbstractJob backupJob = (AbstractJob) context.getBean(BACKUP_JOB_NAME);
+            if (backupJob != null) {
+                this.setTotalNumberOfBackupSteps(backupJob.getStepNames().size());
+            }
+
+            AbstractJob restoreJob = (AbstractJob) context.getBean(BACKUP_JOB_NAME);
+            if (restoreJob != null) {
+                this.setTotalNumberOfRestoreSteps(restoreJob.getStepNames().size());
+            }
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Could not fully configure the Backup Facade!", e);
+        }
+    }
+
+    protected String getItemName(XStreamPersister xp, Class clazz) {
+        return xp.getClassAliasingMapper().serializedClass(clazz);
+    }
+
+    /**
+     * @return
+     * @throws IOException
+     * 
+     */
+    public BackupExecutionAdapter runBackupAsync(final Resource archiveFile,
+            final boolean overwrite, final Filter filter, final Hints params) throws IOException {
+        // Check if archiveFile exists
+        if (archiveFile.file().exists()) {
+            if (!overwrite && FileUtils.sizeOf(archiveFile.file()) > 0) {
+                // Unless the user explicitly wants to overwrite the archiveFile, throw an exception whenever it already exists
+                throw new IOException(
+                        "The target archive file already exists. Use 'overwrite=TRUE' if you want to overwrite it.");
+            } else {
+                FileUtils.forceDelete(archiveFile.file());
+            }
+        } else {
+            // Make sure the parent path exists
+            if (!archiveFile.file().getParentFile().exists()) {
+                try {
+                    archiveFile.file().getParentFile().mkdirs();
+                } finally {
+                    if (!archiveFile.file().getParentFile().exists()) {
+                        throw new IOException("The path to target archive file is unreachable.");
+                    }
+                }
+            }
+        }
+
+        // Initialize ZIP
+        FileUtils.touch(archiveFile.file());
+
+        // Write flat files into a temporary folder
+        Resource tmpDir = BackupUtils.tmpDir();
+
+        // Fill Job Parameters
+        JobParametersBuilder paramsBuilder = new JobParametersBuilder();
+
+        if (filter != null) {
+            paramsBuilder.addString("filter", ECQL.toCQL(filter));
+        }
+
+        paramsBuilder
+                .addString(PARAM_OUTPUT_FILE_PATH,
+                        BackupUtils.getArchiveURLProtocol(tmpDir) + tmpDir.path())
+                .addLong(PARAM_TIME, System.currentTimeMillis());
+
+        parseParams(params, paramsBuilder);
+
+        JobParameters jobParameters = paramsBuilder.toJobParameters();
+
+        // Send Execution Signal
+        BackupExecutionAdapter backupExecution;
+        try {
+            while (!(getRestoreRunningExecutions().isEmpty()
```

Comment:
Actually I forced the run of only 1 execution at a time. The cycle waits a bit for a new execution slot and dies after a timeout (which it is now hardcoded but should be configurable in the future).

Notice that the method getRestoreRunningExecutions() contains only the currently acive jobs (not finished and not stopped or failed)